app/connectors/TaxEnrolmentsConnector.scala

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import audit.Auditable
+import config.{BusinessCustomerFrontendAuditConnector, WSHttp}
+import metrics.{Metrics, MetricsEnum}
+import models._
+import play.api.Logger
+import play.api.http.Status._
+import play.api.libs.json.{JsValue, Json}
+import uk.gov.hmrc.http._
+import uk.gov.hmrc.play.audit.model.{Audit, EventTypes}
+import uk.gov.hmrc.play.config.{AppName, ServicesConfig}
+import utils.GovernmentGatewayConstants
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+object TaxEnrolmentsConnector extends TaxEnrolmentsConnector {
+  val audit: Audit = new Audit(AppName.appName, BusinessCustomerFrontendAuditConnector)
+  val appName: String = AppName.appName
+  val metrics = Metrics
+  val serviceUrl = baseUrl("tax-enrolments")
+  val enrolmentUrl = s"$serviceUrl/tax-enrolments"
+  val http: CoreGet with CorePost = WSHttp
+}
+
+trait TaxEnrolmentsConnector extends ServicesConfig with RawResponseReads with Auditable {
+
+  def serviceUrl: String
+
+  def enrolmentUrl: String
+
+  def http: CoreGet with CorePost
+
+  def metrics: Metrics
+
+  def enrol(enrolRequest: NewEnrolRequest, groupId: String, arn: String)(implicit hc: HeaderCarrier): Future[HttpResponse] = {
+
+    val enrolmentKey = s"${GovernmentGatewayConstants.KnownFactsAgentServiceName}~${GovernmentGatewayConstants.KnownFactsAgentRefNo}~$arn"
+    def createUrl = s"""$enrolmentUrl/groups/$groupId/enrolments/$enrolmentKey"""
+
+    val jsonData = Json.toJson(enrolRequest)
+    val postUrl = createUrl
+
+    val timerContext = metrics.startTimer(MetricsEnum.GG_AGENT_ENROL)
+    http.POST[JsValue, HttpResponse](postUrl, jsonData) map { response =>
```

Comment:
This method is longer than 50 lines so get warning. In awrs-frontend I separated the response status checking into another method to get rid of this warning.